### ISSUE TITLE
Use $LS_COLORS to colorize file completions

### ISSUE THREAD

**Issue by nhooyr** [fish-shell/fish-shell#5939]
When I use `ls`, I get coloured output like so:

<img width="682" alt="Screen Shot 2019-06-14 at 2 18 46 AM" src="https://user-images.githubusercontent.com/10180857/59487478-ced06a00-8e4a-11e9-8b3b-98c3842579ea.png">

The colours are informed by $LS_COLORS.

When I complete in the same directory with fish, it looks like:

<img width="682" alt="Screen Shot 2019-06-14 at 2 18 55 AM" src="https://user-images.githubusercontent.com/10180857/59487495-ddb71c80-8e4a-11e9-962b-28b01329a545.png">

No colors :(

Bash supports this via `colored-stats` in ~/.inputrc.

Another nice feature bash supports that I think is a good fit for fish is the `visible-stats` toggle in ~/.inputrc. Fish kinda supports this with the trailing `/` on directories but bash adds a `*` for executables and everything `ls` supports with the `--indicator-style=classify` option.

**Comment by zanchey** [comment]
I think this is a duplicate of #1599.

**Comment by nhooyr** [comment]
Partially, I'll rename to only be about visible-stats.

**Comment by faho** [comment]
One issue here: Unlike the trailing "/" for directories, we couldn't *use* the `*` from the completion. We don't have something like that, currently, so it would require a bunch of work.

Adding descriptions would be much easier, but it would probably be much too wordy.

(Also IIRC there's a concern here with bad e.g. network filesystems and the additional calls per-file that are required)

**Comment by nhooyr** [comment]
> One issue here: Unlike the trailing "/" for directories, we couldn't use the * from the completion. We don't have something like that, currently, so it would require a bunch of work.

> Adding descriptions would be much easier, but it would probably be much too wordy.

I don't think a description would be terrible but I do think we should try to match `ls`'s output, would be more consistent.

> (Also IIRC there's a concern here with bad e.g. network filesystems and the additional calls per-file that are required)

I doubt it will be significant. `ls` does it all the time.

This reminds me of my comment in https://github.com/fish-shell/fish-shell/issues/3253#issuecomment-310796083

What do you think about modifying the completion of directories to also not insert the trailing slash but show it, just like bash? You said you were thinking about it in https://github.com/fish-shell/fish-shell/issues/3253#issuecomment-233915962 but then closed the issue.

I 100% agree with https://github.com/fish-shell/fish-shell/issues/3253#issuecomment-242490939 that almost all the time, I don't actually want the trailing slash.
